pkg/solana/logpoller/job_get_slots_for_addr.go

```diff
@@ -0,0 +1,115 @@
+package logpoller
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gagliardetto/solana-go"
+	"github.com/gagliardetto/solana-go/rpc"
+
+	"github.com/smartcontractkit/chainlink-solana/pkg/solana/logpoller/worker"
+)
+
+var _ worker.Job = (*getSlotsForAddressJob)(nil)
+
+// getSlotsForAddressJob - identifies slots that contain transactions for specified address in range [from, to] and
+// calls storeSlot for each. If single request was not sufficient to identify all slots - schedules a new job. Channel
+// returned by Done() will be closed only when all jobs are done.
+type getSlotsForAddressJob struct {
+	address   PublicKey
+	beforeSig solana.Signature
+	from, to  uint64
+
+	client RPCClient
+
+	storeSlot func(slot uint64)
+	done      chan struct{}
+	workers   WorkerGroup
+}
+
+func newGetSlotsForAddress(client RPCClient, workers WorkerGroup, storeSlot func(uint64), address PublicKey, from, to uint64) *getSlotsForAddressJob {
+	return &getSlotsForAddressJob{
+		address:   address,
+		client:    client,
+		from:      from,
+		to:        to,
+		storeSlot: storeSlot,
+		workers:   workers,
+		done:      make(chan struct{}),
+	}
+}
+
+func (f *getSlotsForAddressJob) String() string {
+	return fmt.Sprintf("getSlotsForAddress: %s, from: %d, to: %d, beforeSig: %s", f.address, f.from, f.to, f.beforeSig)
+}
+
+func (f *getSlotsForAddressJob) Done() <-chan struct{} {
+	return f.done
+}
+
+func (f *getSlotsForAddressJob) Run(ctx context.Context) error {
+	isDone, err := f.run(ctx)
+	if err != nil {
+		return err
+	}
+
+	if isDone {
+		close(f.done)
+	}
```

Comment:
Is done handled this way because of retries?

Reply:
Yes, if there is an error and job will be retried we should not signal that job was done